dsl/src/test/kotlin/com/chutneytesting/kotlin/synchronize/ScenarioSynchronizeServiceTest.kt

```diff
@@ -155,6 +159,49 @@ class ScenarioSynchronizeServiceTest : HttpTestBase() {
         assertThat(requestJson).containsIgnoringWhitespaces("KOTLIN")
     }
 
+    @Test
+    fun should_create_scenario_with_explicit_id() {
+
+        // Given
+        val scenario = ChutneyScenario(123, "title", "description")
+        val bodyRequest = mapOf("content" to "{\r\n  \"title\": \"title\",\r\n  \"description\": \"description\"\r\n}\r\n",
+                "id" to "123",
+                "title" to "title",
+                "description" to "description",
+                "tags" to listOf("KOTLIN"))
+        mockServer
+                .`when`(
+                        HttpRequest.request()
+                                .withPath("/api/scenario/v2/raw")
+                                .withBody(JsonBody(JSONObject(bodyRequest).toString()))
+                )
+                .respond(
+                        HttpResponse.response()
+                                .withBody("123")
+                )
+
+        mockServer
+                .`when`(
+                        HttpRequest.request()
+                                .withPath("/api/scenario/v2/raw/123")
+                )
+                .respond(
+                        HttpResponse.notFoundResponse()
+                )
+        val chutneyServerInfo = ChutneyServerInfo(
+                url,
+                "aUser",
+                "aPassword"
+        )
+
+        // When
+        val resultId = ChutneyServerServiceImpl.createOrUpdateJsonScenario(chutneyServerInfo, scenario)
+
+        // Then
+        Assertions.assertThat(resultId).isEqualTo(123)
```

Comment:
I feel uneasy with this assertion as this is what the mock has been told to do.
I think the most important here is the assertion, hidden in the mock payload filter, that is the real assertion of this test.

So, either rename bodyRequest in expectedBodyRequest and assert on the mock server expectation id, OR assert the body afterwards with `mockServer.retrieveRecordedRequests`.